src/state-selector.ts

```diff
@@ -181,7 +175,8 @@ export class StateSelector<T> {
                         }),
                         tap(() => this.clearPropertyUpdater(propertyName, pendingRequest$)), // Remove the pending request
                         mergeMap(() => this.state$.pipe(take(1))), // Get the newly updated Session
-                        shareReplay(1)
+                        publishReplay(1),
+                        refCount()
```

Comment:
I'm not sure why this change was required to make pending requests cancelable (theoretically they should have been even with `shareReplay(1)`, but I've tested this and confirmed it works.